h2o-algos/src/main/java/hex/tree/TreeHandler.java

```diff
@@ -0,0 +1,172 @@
+
+        args.left_children = compressedTreeFormat.leftChildren;
+        args.right_children = compressedTreeFormat.rightChildren;
+        args.root_node_number = compressedTreeFormat.rootNodeNumber;
+        return args;
+    }
+
+
+    /**
+     * @param args An instance of {@link TreeV3} input arguments to validate
+     */
+    private void validateArgs(TreeV3 args) {
+        if (args.tree_number < 0) throw new IllegalArgumentException("Tree number must be greater than 0.");
+        if (args.tree_class < 0) throw new IllegalArgumentException("Tree class must be greater than 0.");
+    }
+
+
+    /**
+     * Convers H2O-3's internal representation of a boosted tree in a form of {@link SharedTreeSubgraph} to a compressed format.
+     *
+     * @param sharedTreeSubgraph An instance of {@link SharedTreeSubgraph} to convert
+     * @return An instance of {@link CompressedTreeFormat} with some attributes possibly empty if suitable. Never null.
+     */
+    static CompressedTreeFormat convertSharedTreeSubgraph(final SharedTreeSubgraph sharedTreeSubgraph) {
+        Objects.requireNonNull(sharedTreeSubgraph);
+
+        final CompressedTreeFormat stf = new CompressedTreeFormat();
+        stf.rootNodeNumber = sharedTreeSubgraph.rootNode.getNodeNumber();
+
+        stf.leftChildren = MemoryManager.malloc4(sharedTreeSubgraph.nodesArray.size());
+        stf.rightChildren = MemoryManager.malloc4(sharedTreeSubgraph.nodesArray.size());
+        stf.thresholds = MemoryManager.malloc4f(sharedTreeSubgraph.nodesArray.size());
+
+        // Create a temporary nodeMap for faster search. Our nodes are not always numbered sequentially,
+        // thus searching in the array of nodes by index is not possible.
+        final Map<Integer, SharedTreeNode> nodeMap = createTreeNodeMap(sharedTreeSubgraph.nodesArray);
+
+        // Set root node's children, there is no guarantee the root node will be number 0
+        stf.rightChildren[0] = sharedTreeSubgraph.rootNode.getRightChild() != null ? sharedTreeSubgraph.rootNode.getRightChild().getNodeNumber() : -1;
+        stf.leftChildren[0] = sharedTreeSubgraph.rootNode.getLeftChild() != null ? sharedTreeSubgraph.rootNode.getLeftChild().getNodeNumber() : -1;
+
+        List<SharedTreeNode> nodesToTraverse = new ArrayList<>();
+        nodesToTraverse.add(sharedTreeSubgraph.rootNode);
+        append(stf.rightChildren, stf.leftChildren, stf.thresholds, nodesToTraverse, -1);
+
+        return stf;
+    }
+
+    /**
+     * Creates a map of {@link SharedTreeNode} where number of the node is the key.
+     *
+     * @param sharedTreeNodes A {@link List} of {@link SharedTreeNode} to create map from
+     * @return An non-thread-safe instance of {@link Map} with node number being the key to a {@link SharedTreeNode}
+     */
+    private static Map<Integer, SharedTreeNode> createTreeNodeMap(List<SharedTreeNode> sharedTreeNodes) {
```

Comment:
redundant function

Reply:
Pointless to have it there, leftover. Should be deleted.

Reply:
Done.